java/src/main/java/ai/rapids/cudf/HostMemoryBuffer.java

```diff
@@ -107,15 +110,22 @@ public static HostMemoryBuffer allocate(long bytes) {
   }
 
   HostMemoryBuffer(long address, long length) {
-    super(address, length, new HostBufferCleaner(address));
+    super(address, length, new HostBufferCleaner(address, length));
+    if (length > 0) {
+      MemoryListener.hostAllocation(length, id);
+    }
```

Comment:
I'd recommend replacing this constructor with one that delegates to the other to reduce the duplication, e.g.:
```java
  HostMemoryBuffer(long address, long length) {
    this(address, length, new HostBufferCleaner(address, length));
  }